google/cloud/bigtable/examples/bigtable_examples_common_test.cc

```diff
@@ -409,6 +409,67 @@ TEST(BigtableExamplesCommon, MakeTableAsyncCommandEntry) {
   EXPECT_EQ(1, call_count);
 }
 
+TEST(BigtableExamplesCommon, MakeInstanceAdminAsyncCommandEntry) {
+  // Pretend we are using the emulator to avoid loading the default
+  // credentials from $HOME, which do not exist when running with Bazel.
+  google::cloud::testing_util::ScopedEnvironment emulator(
+      "BIGTABLE_EMULATOR_HOST", "localhost:9090");
+
+  int call_count = 0;
+  auto command = [&call_count](bigtable::InstanceAdmin const&,
+                               CompletionQueue const&,
+                               std::vector<std::string> const& argv) {
+    ++call_count;
+    ASSERT_EQ(2, argv.size());
+    EXPECT_EQ("a", argv[0]);
+    EXPECT_EQ("b", argv[1]);
+  };
+  auto const actual = MakeCommandEntry("command-name", {"foo", "bar"}, command);
+  EXPECT_EQ("command-name", actual.first);
+  EXPECT_THROW(
+      try { actual.second({}); } catch (Usage const& ex) {
+        EXPECT_THAT(ex.what(), HasSubstr("command-name"));
+        EXPECT_THAT(ex.what(), HasSubstr("foo"));
+        EXPECT_THAT(ex.what(), HasSubstr("bar"));
+        throw;
+      },
+      Usage);
+
+  ASSERT_NO_FATAL_FAILURE(actual.second({"unused-project", "a", "b"}));
+  EXPECT_EQ(1, call_count);
+}
+
+TEST(BigtableExamplesCommon, MakeTableAdminAsyncCommandEntry) {
+  // Pretend we are using the emulator to avoid loading the default
+  // credentials from $HOME, which do not exist when running with Bazel.
+  google::cloud::testing_util::ScopedEnvironment emulator(
+      "BIGTABLE_EMULATOR_HOST", "localhost:9090");
+
+  int call_count = 0;
+  auto command = [&call_count](bigtable::TableAdmin const&,
+                               CompletionQueue const&,
+                               std::vector<std::string> const& argv) {
+    ++call_count;
+    ASSERT_EQ(2, argv.size());
+    EXPECT_EQ("a", argv[0]);
+    EXPECT_EQ("b", argv[1]);
+  };
+  auto const actual = MakeCommandEntry("command-name", {"foo", "bar"}, command);
+  EXPECT_EQ("command-name", actual.first);
+  EXPECT_THROW(
```

Comment:
Do we need to be wrapping these tests that throw in `GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS` so that they can be run in environments with exceptions disabled?

Reply:
No. The examples do not compile without exceptions (we actually throw on error to keep the code simple).  I don't see the point of complicating the code to support examples to work in environments without exceptions.